test/src/e2e.dynval/dv.n-1.test.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import * as chai from "chai";
+import { expect } from "chai";
+import * as chaiAsPromised from "chai-as-promised";
+import * as stake from "codechain-stakeholder-sdk";
+import { SDK } from "codechain-sdk";
+import "mocha";
+
+import { validators as originalValidators } from "../../tendermint.dynval/constants";
+import { PromiseExpect } from "../helper/promise";
+import { withNodes } from "./setup";
+
+chai.use(chaiAsPromised);
+
+const alice = originalValidators[0];
+const otherDynValidators = originalValidators.slice(1, 8);
+const allDynValidators = [alice, ...otherDynValidators];
+
+describe("Dynamic Validator N -> N-1", function() {
+    this.timeout(0);
+
+    const promiseExpect = new PromiseExpect();
+    const termSeconds = 20;
+    const margin = 1.2;
+
+    async function aliceDropOutCheck(sdk: SDK) {
+        const blockNumber = await sdk.rpc.chain.getBestBlockNumber();
+        const termMedata = await stake.getTermMetadata(sdk, blockNumber);
+        const currentTermInitialBlockNumber =
+            termMedata!.lastTermFinishedBlockNumber + 1;
+        const validatorsAfter = (await stake.getPossibleAuthors(
+            sdk,
+            currentTermInitialBlockNumber
+        ))!.map(platformAddr => platformAddr.toString());
+
+        expect(termMedata!.currentTermId).to.be.equals(2);
+        expect(validatorsAfter.length).to.be.equals(otherDynValidators.length);
+        expect(validatorsAfter).not.to.includes(
+            alice.platformAddress.toString()
+        );
+        expect(
+            validatorsAfter!.map(platformAddr => platformAddr.toString())
```

Comment:
You don't need `toString` here. Elements of `validatorsAfter` are already converted to a string.

Reply:
Thank you